cpp/src/arrow/engine/substrait/expression_internal.h

```diff
@@ -39,11 +39,6 @@ ARROW_ENGINE_EXPORT
 Result<FieldRef> DirectReferenceFromProto(const substrait::Expression::FieldReference*,
                                           const ExtensionSet&, const ConversionOptions&);
 
-ARROW_ENGINE_EXPORT
```

Comment:
Per this comment:
https://github.com/apache/arrow/pull/34627#discussion_r1152715709

I noticed and removed it here. Although unrelated to the original PR, I figured this is small enough change to piggy back here. If there are objection I can revert this change and put in separate PR.